### ISSUE TITLE
Remote network device unreachable after upgrade from 2019.10.0 to 2020.3.0

### ISSUE THREAD

**Issue by dpizzle** [mirceaulinic/salt-sproxy#135]
**Describe the bug**
After upgrading from `2019.10.0` to `2020.3.0`, I am unable to reach Juniper devices using the napalm module. The devices are listed in the salt pillar file as before.

**Steps To Reproduce**

```
$ cat /etc/salt/master

pillar_roots:
  base:
    - /etc/salt/pillar

$ cat /etc/salt/pillar/top.sls

base: 
  test_switch:
    - napalm_junos

$ cat /etc/salt/pillar/napalm_junos.sls

proxy:
  proxytype: napalm
  driver: junos
  host: {{ grains['id'] }}
  username: testuser
  passwd: testpass
```

```
$ salt-sproxy 'test_switch' test.ping
test_switch:
    Minion did not return. [No response]
ERROR: Minions returned with non-zero exit code
```

Replacing `{{ grains['id'] }}` with `test_switch` in the `napalm_junos.sls` file didn't resolve the issue.

**Versions Report**

```
salt-sproxy 2020.3.0
/etc/salt # salt-sproxy -V
Salt Version:
           Salt: 3000.3
    Salt SProxy: 2020.3.0

Dependency Versions:
        Ansible: Not Installed
           cffi: 1.14.0
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.10
     junos-eznc: 2.2.1
       jxmlease: 1.0.1
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
         NAPALM: 2.5.0
       ncclient: 0.6.7
        Netmiko: 2.4.2
       paramiko: 2.7.1
      pycparser: 2.20
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pyeapi: 0.8.3
         pygit2: Not Installed
       PyNetBox: Not Installed
          PyNSO: Not Installed
         Python: 3.6.9 (default, Oct 17 2019, 11:10:22)
   python-gnupg: Not Installed
         PyYAML: 3.13
          PyZMQ: 19.0.1
            scp: 0.13.2
          smmap: Not Installed
        textfsm: 1.1.0
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.2

System Versions:
           dist:
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.el7.x86_64
         system: Linux
        version: Not Installed
```

**Comment by mirceaulinic** [comment]
Hi @dpizzle. Are you still seeing this issue? I've just tried with salt-sproxy 2020.3.0 and Salt 3000.3 and I'm not able to reproduce this. Would you be able to run with `salt-sproxy 'test_switch' test.ping -l debug` and let share the sanitized output (or maybe you'll notice anything that'll point you in the right direction). Cheers!

**Comment by dpizzle** [comment]
Hi @mirceaulinic, from the debug output, it appears that communication with the switch is working, however towards the end, I see this line `[DEBUG   ] [host test_switch session-id 57665] parsing netconf v1.0` repeat and then the process being terminated.

````
[INFO    ] [host test_switch session-id 57621] Requesting 'ExecuteRpc'
[DEBUG   ] [host test_switch session-id 57621] queueing <?xml version="1.0" encoding="UTF-8"?><nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:045e8c14-f829-4caa-a759-31e1e3804672"><get-interface-information/></nc:rpc>
[DEBUG   ] [host test_switch session-id 57621] Sync request, will wait for timeout=60
[DEBUG   ] [host test_switch session-id 57621] Sending message
[INFO    ] [host test_switch session-id 57621] Sending:
<?xml version="1.0" encoding="UTF-8"?><nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:045e8c14-f829-4caa-a759-31e1e3804672"><get-interface-information/></nc:rpc>]]>]]>
[DEBUG   ] [host test_switch session-id 57621] parsing netconf v1.0
[DEBUG   ] [host test_switch session-id 57665] parsing netconf v1.0
[DEBUG   ] [host test_switch session-id 57665] parsing netconf v1.0
<-- removed duplicate lines -->
[INFO    ] Terminating the process for test_switch, as it didn't reply within 5 seconds
[DEBUG   ] LazyLoaded nested.output
````

**Comment by mirceaulinic** [comment]
The `parsing netconf v1.0` messages are generated by ncclient, and when you see many it usually means the device is slowly replying. Have you tried increasing the execution timeout? (either `timeout: <value>` or passing `-t` option on the CLI, e.g., `salt-sproxy 'test_switch' test.ping -t 60`). The default timeout value is 5 seconds (maybe we should increase this value?), which is why the `Terminating the process for test_switch, as it didn't reply within 5 seconds` message.

The reasoning you may have not seen this on 2019.10.0 was that the timeout wasn't actually properly being handled, and it was fixed in 2020.3.0.

**Comment by dpizzle** [comment]
Using the 60 second timeout worked as expected and it sounds like a better default value.

The help information still mentions 60 seconds -

```
  -t TIMEOUT, --timeout=TIMEOUT
                        Change the timeout, if applicable, for the running
                        command (in seconds). Default: 60.
```

Thank you.

**Comment by mirceaulinic** [comment]
Thanks for confirming @dpizzle. Yes, I've noticed that, should be fixed via #144.
